controllers/dashboard-routes.js

```diff
@@ -0,0 +1,184 @@
+const router = require("express").Router();
+const sequelize = require("../config/connection");
+
+const { User, Post, Comment } = require("../models");
+const { post } = require("./home-routes");
+const withAuth = require("../utils/auth");
+const { restore } = require("../models/Post");
+
+// find all users posts
+router.get("/", async (req, res) => {
+  if (req.session && req.session.loggedIn) {
+    try {
+      const allPosts = await Post.findAll({
+        where: {
+          user_id: req.session.userId,
+        },
+        attributes: ["id", "title", "content", "created_at"],
+        order: [["created_at", "DESC"]],
+        include: [
+          {
+            model: Comment,
+            attributes: [
+              "id",
+              "comment_text",
+              "post_id",
+              "user_id",
+              "created_at",
+            ],
+            include: {
+              model: User,
+              attributes: ["username"],
+            },
+          },
+          {
+            model: User,
+            attributes: ["username"],
+          },
+        ],
+      });
+      // response
+      const posts = allPosts.map((post) => post.get({ plain: true }));
+      // res.render("dashboard", { posts, loggedIn: true });
+      res.render("dashboard", { posts, loggedIn: req.session.loggedIn });
+    } catch (error) {
+      console.log(error);
+      res.status(500).json("Error retrieving posts data from database.");
+    }
+  } else {
+    res.redirect("/login");
```

Comment:
this should ideally be handled through a middleware - let us know if it's something that would be helpful to cover in a workshop.